TypeTheory/ALV2/]

```diff
@@ -0,0 +1,829 @@
+    : functor_compax weak_reluniv_functor_data.
+  Proof.
+    intros a b c f g.
+    use gen_reluniv_mor_eq.
+    - etrans. apply functor_comp. apply idpath.
+    - etrans. apply functor_comp. apply idpath.
+  Defined.
+
+  Definition weak_reluniv_is_functor
+    : is_functor weak_reluniv_functor_data
+    := (weak_reluniv_functor_idax ,, weak_reluniv_functor_compax).
+
+  Definition weak_reluniv_functor
+    : functor (weak_reluniv_cat J) (weak_reluniv_cat J')
+    := make_functor
+         weak_reluniv_functor_data
+         weak_reluniv_is_functor.
+
+  Definition weak_reluniv_functor_is_catiso
+    : is_catiso weak_reluniv_functor.
+  Proof.
+    use tpair.
+    - intros u1 u2.
+      apply weq_weak_reluniv_mor_J_to_J'.
+    - apply weq_weak_relative_universe_transfer.
+  Defined.
+
+  Local Definition relu_J_to_relu_J'
+        (C'_univ : is_univalent C')
+    : functor (reluniv_cat J) (reluniv_cat J')
+    := reluniv_functor_with_ess_surj
+         _ _ _ _ _ _ _ _ _ α_is_iso
+         S_pb C'_univ ff_J' S_full R_es.
+
+  Definition weak_relu_square_commutes_on_mor
+        (C'_univ : is_univalent C')
+        (u1 u2 : reluniv_cat J)
+        (mor : reluniv_cat J ⟦ u1, u2 ⟧)
+    : pr21 (relu_J_to_relu_J' C'_univ ∙ weak_from_reluniv_functor J') u1 u2 mor
+    = pr21 (weak_from_reluniv_functor J ∙ weak_reluniv_functor) u1 u2 mor.
+  Proof.
+    use gen_reluniv_mor_eq.
+    - apply idpath.
+    - apply idpath.
+  Defined.
+
+  Definition weak_relu_square_nat_trans
+        (C'_univ : is_univalent C')
+    : nat_trans
+        (relu_J_to_relu_J' C'_univ ∙ weak_from_reluniv_functor J')
+        (weak_from_reluniv_functor J ∙ weak_reluniv_functor).
+  Proof.
+    use tpair.
+    + intros u.
+      use tpair.
+      * use tpair.
+        -- apply identity.
+        -- apply identity.
+      * unfold is_gen_reluniv_mor.
+        cbn.
+        etrans. apply id_left.
+        apply pathsinv0, id_right.
+    + intros u1 u2 mor.
+      use gen_reluniv_mor_eq.
+      * simpl.
+        etrans. apply id_right.
+        apply pathsinv0, id_left.
+      * simpl.
+        etrans. apply id_right.
+        apply pathsinv0, id_left.
+  Defined.
+
+  Definition weak_relu_square_nat_trans_is_nat_iso
+        (C'_univ : is_univalent C')
+    : is_nat_iso (weak_relu_square_nat_trans C'_univ).
+  Proof.
+    intros u1 wu1.
+    use isweq_iso.
+    - intros mor.
+      apply mor.
+    - intros mor.
+      use id_left. (* FIXME: works, but slow for some reason *)
+    - intros mor.
+      use id_left. (* FIXME: works, but slow for some reason *)
+  Defined.
+    
+  Definition weak_relu_square_commutes
+        (C'_univ : is_univalent C')
+    : nat_iso
+        (relu_J_to_relu_J' C'_univ ∙ weak_from_reluniv_functor J')
+        (weak_from_reluniv_functor J ∙ weak_reluniv_functor).
+  Proof.
+    use tpair.
+    - apply weak_relu_square_nat_trans.
+    - apply weak_relu_square_nat_trans_is_nat_iso.
+  Defined.
+
+  Definition weak_relu_square_commutes_strictly
```

Comment:
I would reserve "strictly" to mean "definitionally":
```suggestion
  Definition weak_relu_square_commutes_identity
```
(I think that this definition is used later, so the code needs to be adapted.)